pkg/testcase/certrotate.go

```diff
@@ -0,0 +1,109 @@
+package testcase
+
+import (
+	"fmt"
+
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestCertRotate() {
+	// Please refer to docs/testing.md for instructions on setup for this test
+
+	cluster := factory.AddCluster(GinkgoT())
+	serverIPs := cluster.ServerIPs
+	agentIPs := cluster.AgentIPs
+	p, _ := shared.GetProduct()
+	var product shared.Product
+	if p == "k3s" {
+		product = shared.K3S
+	} else {
+		product = shared.RKE2
+	}
+
+	certRotate(product, serverIPs)
+
+	product.ManageClusterService(shared.Restart, shared.Agent, agentIPs)
```

Comment:
you need to handle error callback here since this returns err

Reply:
fixed